src/helper-modules/govuk-notify/govuk-notify.service.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  BadRequestException,
+  ForbiddenException,
+  Injectable,
+  InternalServerErrorException,
+  UnauthorizedException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import { PinoLogger } from 'nestjs-pino';
+import { NotifyClient } from 'notifications-node-client';
+
+import { PostEmailsRequestItemDto } from '../../modules/emails/dto/post-emails-request.dto';
+import { PostEmailsResponseDto } from './dto/post-emails-response.dto';
+
+@Injectable()
+export class GovukNotifyService {
+  constructor(private readonly logger: PinoLogger) {}
+
+  /**
+   * Send email to one recipient using GOV.UK template id
+   * @param {String} govUkNotifyKey
+   * @param {PostEmailsRequestItemDto} post email request item
+   *
+   * @returns {Promise.<PostEmailsResponseDto>} GOV.UK Notify response
+   *
+   * @throws {BadRequestException}
+   * @throws {UnauthorizedException}
+   * @throws {ForbiddenException}
+   * @throws {Error}
+   * @throws {UnprocessableEntityException}
+   * @throws {InternalServerErrorException}
+   */
+  async sendEmail(
+    govUkNotifyKey: string,
+    email: PostEmailsRequestItemDto,
+  ): Promise<
+    | PostEmailsResponseDto
+    | BadRequestException
+    | UnauthorizedException
+    | ForbiddenException
+    | Error
+    | UnprocessableEntityException
+    | InternalServerErrorException
+  > {
```

Comment:
```suggestion
  ): Promise<PostEmailsResponseDto> {
```

Reply:
From my understanding, this union type implies that this function can *return* one of these error types, when in fact this isn't the case, as it throws them. This is less important here as we don't try to access any properties of `PostEmailsResponseDto` in any downstream functions, but if, for example, we did `const result = await this.govukNotifyService.sendEmail(govUkNotifyKey, postEmailRequestItem)` in the `EmailsService` and then we wanted to access `result.data`, TypeScript would complain at us that `Property 'data' does not exist on type 'Error'.`, even though it is perfectly safe for us to access this property as `const result = await this.govukNotifyService.sendEmail(govUkNotifyKey, postEmailRequestItem)` would *throw*, rather than return an `Error`.

Reply:
Strangely typescript didn't complain, but I removed returned exception types, I was aware that throwing and returning are different things.